pkg/ibmvpc/routing_tables_analysis.go

```diff
@@ -0,0 +1,337 @@
+		case ResourceTypePublicGateway:
+			res.pgwList = append(res.pgwList, router.(*PublicGateway))
+		case ResourceTypeFloatingIP:
+			res.fipList = append(res.fipList, router.(*FloatingIP))
+		}
+	}
+	return res
+}
+
+func newRoutingTable(routes []*route, vpcConfig *vpcmodel.VPCConfig, vpc *VPC) *routingTable {
+	if vpcConfig == nil {
+		vpcConfig = &vpcmodel.VPCConfig{}
+	}
+	res := &routingTable{routesList: routes}
+	res.computeDisjointRouting()
+	res.implicitRT = &systemImplicitRT{vpc: vpc, config: configFromVPCConfig(vpcConfig)}
+	return res
+}
+
+type routingTable struct {
+	//nolint:unused // to be used later
+	name string // should implement VPCResourceIntf instead
+
+	// nextHops is a map from disjoint ip-blocks, after considering route preferences and actions
+	nextHops map[*ipblock.IPBlock]*ipblock.IPBlock // delivered ip-blocks
+
+	droppedDestinations *ipblock.IPBlock // union of all ip-ranges for dropped destinations
+
+	delegatedDestinations *ipblock.IPBlock // union of all ip-ranges for delegated destinations
+
+	// routesList is the list of routes that were added for this routing table
+	// it is used to compute the fields above
+	routesList []*route
+	implicitRT *systemImplicitRT
+}
+
+func (rt *routingTable) disjointRoutingStr() string {
+	lines := []string{}
+	for dest, nextHop := range rt.nextHops {
+		lines = append(lines, fmt.Sprintf("%s -> %s", dest.ToIPRanges(), nextHop.ToIPAddressString()))
+	}
+	for _, droppedDest := range rt.droppedDestinations.ToCidrList() {
+		lines = append(lines, fmt.Sprintf("%s -> drop", droppedDest))
+	}
+	for _, delegatedDest := range rt.delegatedDestinations.ToCidrList() {
+		lines = append(lines, fmt.Sprintf("%s -> delegate", delegatedDest))
+	}
+	slices.Sort(lines)
+	return strings.Join(lines, "\n")
+}
+
+func (rt *routingTable) computeRoutingForDisjointDest(disjointDest *ipblock.IPBlock) {
+	// find the relevant rule from the sorted list of rules
+	for _, routeRule := range rt.routesList {
+		if disjointDest.ContainedIn(routeRule.destIPBlock) {
+			log.Default().Printf("%s contained in %s\n", disjointDest.ToIPRanges(), routeRule.destination)
+			switch routeRule.action {
+			case deliver:
+				rt.nextHops[disjointDest] = routeRule.nextHopIPBlock
+				log.Default().Printf("set next hop for %s as %s\n", disjointDest.ToIPRanges(), routeRule.nextHop)
+				return // skip next rules, move to the next disjoint dest
+			case drop:
+				rt.droppedDestinations = rt.droppedDestinations.Union(disjointDest)
+				log.Default().Printf("set %s as drop\n", disjointDest.ToIPRanges())
+				return // skip next rules, move to the next disjoint dest
+			case delegate:
+				rt.delegatedDestinations = rt.delegatedDestinations.Union(disjointDest)
+				log.Default().Printf("set %s as delegate\n", disjointDest.ToIPRanges())
+				return // skip next rules, move to the next disjoint dest
+			}
+		}
+	}
+}
+
+func (rt *routingTable) getPath(src vpcmodel.Node, dest *ipblock.IPBlock) vpcmodel.Path {
```

Comment:
It will make more sense if `getPath` appear after `computeDisjointRouting`

Reply:
done